### PR TITLE
remove user agent from isMobile

### PR DESCRIPTION
**Pull request by ejnshtein**
Move isMobile to config reducer and use it via connecting function in components.

### CODE REVIEW

**Comment by ejnshtein** [comment]
Currently unable to test new behavior on iPhone and iPad in Chrome Devtools.
The client hints return empty values.
Screenshot 2020-09-16 at 18 19 30
 
However, Pixel 2 emulation works properly.
Screenshot 2020-09-16 at 18 20 23

**Review by yeegor — CHANGES_REQUESTED** [review]

File: src/app/store/Config/Config.action.js [hunk 1]
```diff
@@ -15,3 +15,11 @@ export const updateConfig = (config) => ({
     type: UPDATE_CONFIG,
     config
 });
+
+export const UPDATE_DEVICE = 'UPDATE_DEVICE';
+
+/** @namespace Store/Config/Action/updateDeviceConfig */
```
[the remark this answers is not in the document]
Comment (ejnshtein) [reply src/app/store/Config/Config.action.js:21]: Should I use UPDATE_CONFIG_DEVICE then?

File: src/app/store/Config/Config.reducer.js [hunk 2]
```diff
@@ -34,7 +34,18 @@ export const getInitialState = () => ({
     countries,
     reviewRatings,
     checkoutAgreements: [],
-    isLoading: true
+    isLoading: true,
+    device: {
+        mobile: true,
```
[the remark this answers is not in the document]
Comment (ejnshtein) [reply src/app/store/Config/Config.reducer.js]: Done.

File: src/app/component/MenuItem/MenuItem.component.js [hunk 3]
```diff
@@ -36,10 +37,11 @@ export class MenuItem extends PureComponent {
     };
 
     renderItemContentImage(icon, itemMods) {
+        const { device } = this.props;
         const { isBanner, isLogo, type } = itemMods;
 
         if (!icon
-            || ((!isMobile.any() && !isMobile.tablet()) && !isBanner && !isLogo)
+            || (!device.isMobile && !isBanner && !isLogo)
```
[the remark this answers is not in the document]
Comment (ejnshtein) [reply src/app/component/MenuItem/MenuItem.component.js:44]: Yeah, will add it.

**Review by yeegor — APPROVED** [review]

**Review by yeegor — CHANGES_REQUESTED** [review]
Clicking related products on PDP breaks the application on mobile (tested on Chrome on Pixel 2 XL, emulated with my mac)

**Comment by ejnshtein** [comment]
Should be fixed now

File: src/app/store/Config/Config.action.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (yeegor) [reply src/app/store/Config/Config.action.js]: It should be updateDevice

File: src/app/component/Router/Router.container.js [hunk 4]
```diff
@@ -117,6 +131,34 @@ export class RouterContainer extends PureComponent {
         }
     }
 
+    componentWillUnmount() {
```
[the remark this answers is not in the document]
Comment (yeegor) [reply src/app/component/Router/Router.container.js:134]: It's been decided that this logic is going to be moved to a separate component

**Review by yeegor — APPROVED** [review]
All the issues are fixed

**Comment by yeegor** [comment]
Remember to reference the issue your PR fixes by: Resolves #716

**Pull request merged**